h5io/_h5io.py

```diff
@@ -423,3 +424,63 @@ def __init__(self):
 
     def __del__(self):
         rmtree(self._path, ignore_errors=True)
+
+
+def _list_file_contents(h5file):
+    h5py = _check_h5py()
```

Comment:
to me it makes sense to do all the type checking / triaging in one place, so I'd be in favor of moving the `h5py.File` check and the check for the `'h5io'` key into the public function.